scalactic.dotty/src/main/scala/org/scalactic/anyvals/NumericString.scala

```diff
@@ -541,7 +541,7 @@ final class NumericString private (val value: String) extends AnyVal {
    * @return         array of strings produced by splitting `NumericString` at every location matching supplied `regex`
    */
   def split(regex: String): Array[String] =
-    value.split(regex)
+    value.split(regex).map(_.nn)
```

Comment:
**Java method returns array**
Java method returns array of type `Array[Null | T]`. We need to map it to `Array[T]`.